### DISCUSSION TITLE
Script Macro: How to post a message to Message panel silently like console.log, instead of alert.

### DISCUSSION THREAD

**Discussion by Lyeebn** [texstudio-org/texstudio#3359] [Q&A] [answered]
I am writing Script Macros, the help document tells:
① alert、information、 warning、critical
    These all generate dialogue boxes, 
    Thus We have to close them afterwards. 
     I am lazy, I don't want to pay an extra click action.
② confirm confirmWarning (No discussed here.)
③ debug(str)
     I don't know what this debug function do, nothing?

We see that, Message panel show the complying status, command line calling and something else silently.
Functions in ① are analogous, they differ just in icons. Personally, I thinks that is redundancy somehow.

It would be nice if one of ① or debug post 'str' to message panel.

**Comment by SwitWu** [comment]
![TexStudioScriptMessage](https://github.com/texstudio-org/texstudio/assets/39337667/ca375d49-f478-46d5-904a-303b6c229cb4)

**Reply by Lyeebn** [reply]
Thank you, my friends. It helps.

 Chrome  of low vision cannot paste img or click the buttons.

**Comment by sunderme** [comment, marked as answer]
```
app.processNotification("test")
```

**Reply by Lyeebn** [reply]
It works!
I think it deserve a place in Help Document.
It's really nice of you. Cannot thanks you too much!

I tried:
  `for (x in app) {app.processNotification(x) ; editor.write(x + '\n');}`
and found many hack APIs. Awesome, Cool.

 `app.processNotification("processNotification( <font style='font-size:25px; background-color:red; color:blue;'> QString Support rich CSS format</font> ) Wawoo! ")`
